validator/keymanager/remote-web3signer/v1/requests.go

```diff
@@ -0,0 +1,262 @@
+package v1
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	validatorpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1/validator-client"
+)
+
+// GetBlockSignRequest maps the request for signing type BLOCK.
+func GetBlockSignRequest(request *validatorpb.SignRequest, genesisValidatorsRoot []byte) (*BlockSignRequest, error) {
+	beaconBlock, ok := request.Object.(*validatorpb.SignRequest_Block)
+	if !ok {
+		return nil, errors.New("failed to cast request object to block")
+	}
+	if beaconBlock == nil {
+		return nil, errors.New("invalid sign request: BeaconBlock is nil")
+	}
+	fork, err := MapForkInfo(request.SigningSlot, genesisValidatorsRoot)
+	if err != nil {
+		return nil, err
+	}
+	beaconBlockBody, err := MapBeaconBlockBody(beaconBlock.Block.Body)
+	if err != nil {
+		return nil, err
+	}
+	return &BlockSignRequest{
+		Type:        "BLOCK",
+		ForkInfo:    fork,
+		SigningRoot: hexutil.Encode(request.SigningRoot),
+		Block: &BeaconBlock{
+			Slot:          fmt.Sprint(beaconBlock.Block.Slot),
+			ProposerIndex: fmt.Sprint(beaconBlock.Block.ProposerIndex),
+			ParentRoot:    hexutil.Encode(beaconBlock.Block.ParentRoot),
+			StateRoot:     hexutil.Encode(beaconBlock.Block.StateRoot),
+			Body:          beaconBlockBody,
+		},
+	}, nil
+}
+
+// GetAggregationSlotSignRequest maps the request for signing type AGGREGATION_SLOT.
+func GetAggregationSlotSignRequest(request *validatorpb.SignRequest, genesisValidatorsRoot []byte) (*AggregationSlotSignRequest, error) {
```

Comment:
Here and in all methods in this file. I wont comment on all as not to spam, but please fix.